### ISSUE TITLE
BuildPackages do not log

### ISSUE THREAD

**Issue by hungaborhorvath** [gap-system/gap#1014]
````BuildPackages.sh```` does not log anywhere, or if yes, then I could not find it. 

Considering the immense output this generates (especially if sometimes a package fails to be built and it refers to some distribution package dependencies), I always run it like this in the ````pkg```` directory so that I can get back to it later: 
````
(../bin/BuildPackages.sh > >(tee ./buildpackages.out) 2> >(tee ./buildpackages.err >&2) ) > ./buildpackages.log 2>&1
````
This creates files ````pkg/buildpackages.out```` with the ````stdout```` output, ````pkg/buildpackages.err```` with the ````stderr```` output, and ````pkg/buildpackages.log```` with the complete output with lines coming properly after each other.

I guess this can (and should) be implemented in some way into ````BuildPackages.sh````

**Comment by hungaborhorvath** [comment]
Also, it would be nice if there was at least an empty line before and after a package build to better read the output.

**Comment by fingolfin** [comment]
I consider `BuildPackages.sh` a stop-gap measure at best anyway. It's better than nothing, but ultimately, it's a bad substitute for a proper package manager...

Anyway, yeah, some logging would be helpful, I guess.

Also, perhaps use color on the terminal output, to highlight messages printed by the script, so that they stand out when scrolling back in the terminal. While we are collecting wisehs for it, or rather a replacement package manager:

- allow building only some packages, or all except some you want to skip (so if one package failed to compile, and you fixed the cause of that, you have to recompile all packages in order to see whether the fix worked).
- allow downloading and installing/compiling newer versions of packages
- also (optionally) rebuild package manuals (to ensure reference links are not broken)

**Comment by ChrisJefferson** [comment]
Yes, to be clear, I made the current `BuildPackages.sh` and it is very horrible. Feel free to make any changes.

One option would be to output all output to a file per package, and on the screen just print which packages built correctly, and which didn't.

**Comment by ChrisJefferson** [comment]
Also seperating building one package from all packages -- I started this one day, but I'm not sure bash is the right language to do it in, for a start it's hard in bash to run another script from the same directory as an already-running script, so splitting into `BuildPackage.sh` and `BuildPackages.sh` is tricky.

**Comment by fingolfin** [comment]
Also to be clear: Despite what I wrote, I don't consider `BuildPackages.sh` to be horrible. It's useful, and much better than what we had before! But of course on the long run, we need something better. But until somebody has the abilitie and time to implement such a better thing, I am glad we have it.

**Comment by hungaborhorvath** [comment]
I am no sure if it is a good idea to log every package into a different file. In that case, where would we save the log files? The ````pkg```` directory is cluttered already as it is, saving into e.g. ````pkg/log```` could make GAP (or ````BuildPackages.sh````) think that ````log```` is a package....

**Comment by alex-konovalov** [comment]
@hungaborhorvath `BuildPackages.sh` ignores directories that do not contain `PackageInfo.g` file, so it will ignore `pkg/log`. Same is for `InitializePackagesInfoRecords`, which, if `PackageInfo.g` ios not found, also recursively goes through subdirectories.

**Comment by hungaborhorvath** [comment]
@alex-konovalov @fingolfin @ChrisJefferson #1023 now should log every package into a separate file.

**Comment by fingolfin** [comment]
Resolved by PR #1023
